collector/lib/ProcessSignalFormatter.h

```diff
@@ -52,6 +56,8 @@ class ProcessSignalFormatter : public ProtoSignalFormatter<sensor::SignalStreamM
   const EventNames& event_names_;
   std::unique_ptr<system_inspector::EventExtractor> event_extractor_;
   ContainerMetadata container_metadata_;
+
+  const CollectorConfig* config_;
```

Comment:
Could we store this as a reference to make it clear it can never be null and it won't be reassigned? (at least for now).
```suggestion
  const CollectorConfig& config_;
```